wcfsetup/install/files/lib/system/session/SessionHandler.class.php

```diff
@@ -201,38 +201,116 @@ public function setCookieSuffix() { }
 	public function setHasValidCookie($hasValidCookie) { }
 	
 	/**
-	 * Returns the session ID stored in the session cookie or `null`.
+	 * Parses the session cookie value, returning an array with the stored fields.
+	 * 
+	 * The return array is guaranteed to have a `sessionId` key.
+	 */
+	private function parseCookie(string $value): array {
+		$length = \mb_strlen($value, '8bit');
+		if ($length < 1) {
+			throw new \InvalidArgumentException(\sprintf(
+				'Expected at least 1 Byte, %d given.',
+				$length
+			));
+		}
+		
+		$version = \unpack('Cversion', $value)['version'];
+		if (!in_array($version, [1], true)) {
+			throw new \InvalidArgumentException(\sprintf(
+				'Unknown version %d',
+				$version
+			));
+		}
+		
+		if ($version === 1) {
+			if (\mb_strlen($value, '8bit') !== 26) {
+				throw new \InvalidArgumentException(\sprintf(
+					'Expected exactly 26 Bytes, %d given.'
```

Comment:
Missing value for `%d`.

Reply:
Indeed. I wanted to refactor that to simply use `$length`, but I must have gotten distracted.